### ISSUE TITLE
Create AWS image with our tools and codebase

### ISSUE THREAD

**Issue by ibeltagy** [bigscience-workshop/Megatron-DeepSpeed#119]
to make it easy for new users to use our setup

**Comment by stas00** [comment]
Could you be a bit more specific, Iz? To run Meg-DS training?

I have the more or less ready AWS image I created for the CI - but I'm definitely not a AWS expert.

Perhaps it'd be better to create a docker image that already includes everything including torch and cuda, etc.?

**Comment by ibeltagy** [comment]
Yes, to run Meg-DS training. Basically doing the steps listed in readme here https://github.com/bigscience-workshop/Megatron-DeepSpeed for them so that they only need to run the `pretrain_*` script.

**Comment by stas00** [comment]
The only problem with this pre-made image is that our components are in flax - e.g. we get fixes in the deepspeed repo, Meg-DS gets changed too and so are transformers if we start using those.

So if you look here, we dynamically install the latest repos:
https://github.com/bigscience-workshop/Megatron-DeepSpeed/blob/c839a8aa30731f71b3738d56009be9668508e366/.github/workflows/main.yml#L109-L113

But otherwise the image is ready already.

I'm glad that it'll have a use as I was about to discard it, as I have to re-do it for GCP now.

**Comment by stas00** [comment]
@philschmid, could you please help me to make this image we made for Megatron-Deepspeed CI somehow available to the wider group? Basically anybody at BigScience. I'm not sure if we have to do something special there, I'm an AWS newbie.

Thank you.

**Comment by philschmid** [comment]
We could create AWS AMI and make it publicly available. This could then be used on EC2 by others.

**Comment by stas00** [comment]
That would be fantastic! Thank you, Philipp!

**Comment by ibeltagy** [comment]
@jaketae can be the first user of the AMI

**Comment by stas00** [comment]
@ibeltagy, is this going to be used on EC2 on user's personal account or some HF account or else?

**Comment by stas00** [comment]
@ibeltagy?
